### DISCUSSION TITLE
orgmode table editor

### DISCUSSION THREAD

**Discussion by chipsenkbeil** [nvim-orgmode/orgmode#743] [Ideas]
Per https://orgmode.org/manual/Built_002din-Table-Editor.html, there is a builtin table editor for orgmode within Emacs.

The only functionality I've noticed with the builtin table functionality is table formatting (thank you for this <3). I've recently been working on my finances a bit in an org document, and could really benefit from some of the table navigation and summation features that Emacs has to make my life easier.

Gave the recommended [vim-table-mode](https://github.com/dhruvasagar/vim-table-mode) plugin a try, but not only did I find it cumbersome to use, I also have a hard time contributing to it as it uses vimscript. Seems like it's quite the old plugin (still maintained, though!).

Curious what your thoughts are @kristijanhusak on the builtin editor functionality for this plugin since it's a standard feature of the core orgmode Emacs plugin. Did I miss extra features? Have you given thought to how you would pursue?

I could eventually tackle an `org-table.nvim` plugin to add some of what I'd want, but didn't want to dive into this without your perspective first.

**Comment by kristijanhusak** [comment]
I'm interested in having the table mode built in. There is some support, as you said, with reformatting and handling new lines while in the table, but there's nothing else. 
If you want, you can start working on some additional stuff. You can put it here https://github.com/nvim-orgmode/orgmode/tree/master/lua/orgmode/files/elements/table so it's grouped.

**Comment by qaptoR** [comment]
@chipsenkbeil have you already started working on this?

I'm an avid user of tables to organize data in emacs, and am highly motivated to help see this feature completed since neovim is now my daily weapon of choice. If you would like help let me know, otherwise we might be doubling up on the tasks being done.

**Reply by chipsenkbeil** [reply]
No, haven't started on this yet :) Feel free to dive in to this one yourself!

**Reply by qaptoR** [reply]
awesome, thanks. just orienting myself in the code... why did I pick such a large project as my staring point for learning plugins 🤦‍♂️

**Comment by qaptoR** [comment]
@chipsenkbeil I felt a little out of my depth jumping straight into the orgmode repo, as I've never written a plugin for neovim before (wrote a couple for obsidian though).

decided to take this opportunity to rewrite vim-table-mode in lua, as I see quite a few projects are rewrites of well known vimscript plugins.

it will be a while before It reaches feature parity, as I personally don't mind using vim-table-mode as is, so the future of `fantableous.nvim` will be slowly developed when I have the time. but here it is anyway if you want to add on to vim-table-mode some much needed features that were missing imo.

https://github.com/qaptoR-nvim/fantableous.nvim

basically implemented move cell, and move column using treesitter. probably very crude implementation, but I'm new to coding in lua. works well enough though.
